### PR TITLE
HDDS-4392. [DOC] Add Recon architecture to docs

### PR DESCRIPTION
**Pull request by vivekratnavel**
## What changes were proposed in this pull request?

Add Recon Architecture and API reference documentation

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4392

## How was this patch tested?

Tested with hugo in local. 

```
cd hadoop-hdds/docs
hugo serve
```

### CODE REVIEW

File: hadoop-hdds/docs/content/concept/Recon.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+{{<figure src="/concept/ReconScmDesign.png" width="800px">}}
+
+<br/>
+
+Recon also acts as a passive SCM for datanodes. When Recon is configured in the
+cluster, all the datanodes register with Recon and send heartbeats, container 
+reports, incremental container reports etc. to Recon similar to SCM. Recon uses
+all the information it gets from datanodes to construct its own copy of SCM rocks db 
+locally. Recon never sends any command to datanodes in response and just acts as
+a passive SCM for faster querying of SCM database.
```
Comment (swagle) [inline hadoop-hdds/docs/content/concept/Recon.md:80]: Minor nit: replace faster querying with allows independently query SCM database.

File: hadoop-hdds/docs/content/concept/Recon.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+<br/>
+
+Recon also acts as a passive SCM for datanodes. When Recon is configured in the
+cluster, all the datanodes register with Recon and send heartbeats, container 
+reports, incremental container reports etc. to Recon similar to SCM. Recon uses
+all the information it gets from datanodes to construct its own copy of SCM rocks db 
+locally. Recon never sends any command to datanodes in response and just acts as
+a passive SCM for faster querying of SCM database.  
+
+The data in local SCM db is consumed by all the SCM db tasks triggered at
```
Comment (swagle) [inline hadoop-hdds/docs/content/concept/Recon.md:82]: Duplication below. Task framework will use all the dbs, from this statement I get the impression that only SCM will be.

File: hadoop-hdds/docs/content/concept/Recon.md (hunk 1 above)
Comment (avijayanhwx) [inline hadoop-hdds/docs/content/concept/Recon.md:80]: Nit. _for faster lookup of SCM metadata_.
(DB lookups are not done in SCM mostly.)

File: hadoop-hdds/docs/content/concept/Recon.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+all the information it gets from datanodes to construct its own copy of SCM rocks db 
+locally. Recon never sends any command to datanodes in response and just acts as
+a passive SCM for faster querying of SCM database.  
+
+The data in local SCM db is consumed by all the SCM db tasks triggered at 
+defined intervals via [Recon Tasks Framework](#tasks-framework).
+
+## <a name="tasks-framework"></a> Tasks Framework
+
+Recon has its own Tasks framework to enable batch processing of data obtained
```
Comment (avijayanhwx) [inline hadoop-hdds/docs/content/concept/Recon.md:87]: Nit. _Task_ framework

File: hadoop-hdds/docs/content/concept/Recon.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+ * The following data is stored in the configured SQL database (default is Derby):
+     * GlobalStats table
+       * A Key -> Value table to store aggregate information like total
+       number of volumes / buckets / keys present in the cluster
+     * FileCountBySize table
+       * Keeps track of the number of files present within a file size range in the cluster
+     * ReconTaskStatus table
+       * Keeps track of the status and last run timestamp of the registered OM and SCM 
+       db tasks in the [Recon Tasks Framework](#tasks-framework)
+     * ContainerHistory table
```
Comment (avijayanhwx) [inline hadoop-hdds/docs/content/concept/Recon.md:143]: Just FYI, the 'ContainerHistory' SQL table will likely be removed in HDDS-4403.

File: hadoop-hdds/docs/content/concept/Recon.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+## Notable configurations
+
+key | default | <div style="width: 300px;">description</div>
+----|---------|------------
+ozone.recon.http-address | 0.0.0.0:9888 | The address and the base port where the Recon web UI will listen on.
+ozone.recon.address | 0.0.0.0:9891 | RPC address of the Recon.
+ozone.recon.db.dir | none | Directory where the Recon Server stores its metadata.
+ozone.recon.om.db.dir | none | Directory where the Recon Server stores its OM snapshot DB.
+ozone.recon.om.snapshot<br>.task.interval.delay | 10m | Interval in MINUTES by Recon to request OM DB Snapshot.
```
Comment (avijayanhwx) [inline hadoop-hdds/docs/content/concept/Recon.md:159]: Interval in MINUTES by Recon to request OM DB Snapshot _/ delta updates_.

**Review by avijayanhwx — COMMENTED** [review]
Thanks for adding this @vivekratnavel. Minor comments inline.

File: hadoop-hdds/docs/content/concept/Recon.md (hunk 4 above)
Comment (vivekratnavel) [reply hadoop-hdds/docs/content/concept/Recon.md:143]: Ack.

**Comment by vivekratnavel** [comment]
@swagle @avijayanhwx Thanks for the review comments! I have addressed them all. Please review again.

**Review by avijayanhwx — APPROVED** [review]

**Pull request merged**